File: app/src/main/java/social/entourage/android/entourage/information/discussion/DateSeparatorViewHolder.java
package social.entourage.android.entourage.information.discussion;

import android.view.View;
import android.widget.TextView;

import social.entourage.android.R;
import social.entourage.android.api.model.TimestampedObject;
import social.entourage.android.base.BaseCardViewHolder;
import social.entourage.android.tools.Utils;

/**
 * Created by mihaiionescu on 15/03/2017.
 */

public class DateSeparatorViewHolder extends BaseCardViewHolder {

    private TextView mTimestampView;

    public DateSeparatorViewHolder(final View view) {
        super(view);
    }

    @Override
    protected void bindFields() {
        mTimestampView = itemView.findViewById(R.id.tic_date_separator_timestamp);
    }

    @Override
    public void populate(final TimestampedObject data) {
        if (data == null || mTimestampView == null) return;
        mTimestampView.setText(Utils.dateAsStringFromNow(data.getTimestamp(), itemView.getContext()));
    }

    public static int getLayoutResource() {
        return R.layout.tour_information_date_separator_card;
    }

}

File: app/src/main/java/social/entourage/android/entourage/information/discussion/DateSeparatorViewHolder.kt
package social.entourage.android.entourage.information.discussion

import android.view.View
import kotlinx.android.synthetic.main.layout_entourage_information_date_separator_card.view.*
import social.entourage.android.R
import social.entourage.android.api.model.TimestampedObject
import social.entourage.android.base.BaseCardViewHolder
import social.entourage.android.tools.Utils

/**
 * Created by mihaiionescu on 15/03/2017.
 */
class DateSeparatorViewHolder(view: View) : BaseCardViewHolder(view) {
    override fun bindFields() {
    }

    override fun populate(data: TimestampedObject) {
        itemView.tic_date_separator_timestamp?.text = Utils.dateAsStringFromNow((data as DateSeparator).timestamp, itemView.context)
    }

    companion object {
        val layoutResource: Int
            get() = R.layout.layout_entourage_information_date_separator_card
    }
}